### ISSUE TITLE
failed streams are not killed

### ISSUE THREAD

**Issue by vasell86** [cyb3rdoc/ez-nvr#12]
When recording fails the old process is not killed

2024-08-24 06:38:59,981 [ERROR]: NVR: Recording for camera garaj_curte2 has stopped unexpectedly!
2024-08-24 06:38:59,997 [INFO]: NVR: Recording restarted for camera garaj_curte2.
2024-08-24 06:39:00,010 [INFO]: NVR: Connection established to garaj_curte2 at 192.168.11.131
2024-08-24 06:39:00,018 [INFO]: NVR: Camera garaj_curte2 initialized.

$ ps -ef | grep ffmpeg | grep curte2

root     2955376 2955329  2 aug23 ?        00:40:07 ffmpeg -hide_banner -y -loglevel error -rtsp_transport tcp -use_wallclock_as_timestamps 1 -i rtsp://XXX@192.168.11.131/stream1 -c copy -f segment -reset_timestamps 1 -segment_time 600 -segment_format mkv -segment_atclocktime 1 -strftime 1 /storage/garaj_curte2/raw/%Y-%m-%dT%H-%M-%S.mkv

root     3327014 2955329  3 06:38 ?        00:05:40 ffmpeg -hide_banner -y -loglevel error -rtsp_transport tcp -use_wallclock_as_timestamps 1 -i rtsp://XXX@192.168.11.131/stream1 -c copy -f segment -reset_timestamps 1 -segment_time 600 -segment_format mkv -segment_atclocktime 1 -strftime 1 /storage/garaj_curte2/raw/%Y-%m-%dT%H-%M-%S.mkv

**Comment by cyb3rdoc** [comment]
I updated the code to clean up the camera process and terminate failed ffmpeg streams. Updated code is in dev branch and docker image uses dev tag. Check dev docker image and let me know if this fixes your issue.

**Comment by 991548267** [comment]
dev分支，rocord失败
2024-12-12 11:25:11,637 [INFO]: NVR: Initializing EZ-NVR...
2024-12-12 11:25:11,661 [INFO]: NVR: Connecting to Camera1 at 10_10_10_215.
2024-12-12 11:25:11,675 [INFO]: NVR: Connecting to Camera2 at 10_10_10_77.
2024-12-12 11:25:11,679 [ERROR]: NVR: Recording for camera Camera1 has stopped unexpectedly!
2024-12-12 11:25:11,680 [ERROR]: NVR: No ffmpeg process found for Camera1.
2024-12-12 11:25:11,708 [INFO]: NVR: Recording restarted for camera Camera1.
2024-12-12 11:25:11,709 [ERROR]: NVR: No connection to Camera2 at 10_10_10_77
2024-12-12 11:25:11,712 [ERROR]: NVR: Recording for camera Camera2 has stopped unexpectedly!
2024-12-12 11:25:11,712 [ERROR]: NVR: No ffmpeg process found for Camera2.
2024-12-12 11:25:11,715 [INFO]: NVR: Recording restarted for camera Camera2.
2024-12-12 11:25:11,718 [ERROR]: NVR: No connection to Camera1 at 10_10_10_215
2024-12-12 11:25:11,722 [ERROR]: NVR: No connection to Camera1 at 10_10_10_215
2024-12-12 11:25:11,727 [ERROR]: NVR: No connection to Camera2 at 10_10_10_77
2024-12-12 11:26:11,717 [ERROR]: NVR: Recording for camera Camera1 has stopped unexpectedly!
2024-12-12 11:26:11,718 [ERROR]: NVR: No ffmpeg process found for Camera1.
2024-12-12 11:26:11,719 [INFO]: NVR: Recording restarted for camera Camera1.
2024-12-12 11:26:11,721 [ERROR]: NVR: Recording for camera Camera2 has stopped unexpectedly!
2024-12-12 11:26:11,722 [ERROR]: NVR: No ffmpeg process found for Camera2.
2024-12-12 11:26:11,724 [INFO]: NVR: Recording restarted for camera Camera2.
2024-12-12 11:26:11,732 [ERROR]: NVR: No connection to Camera1 at 10_10_10_215
2024-12-12 11:26:11,738 [ERROR]: NVR: No connection to Camera2 at 10_10_10_77
2024-12-12 11:27:11,727 [ERROR]: NVR: Recording for camera Camera1 has stopped unexpectedly!
2024-12-12 11:27:11,727 [ERROR]: NVR: No ffmpeg process found for Camera1.
2024-12-12 11:27:11,729 [INFO]: NVR: Recording restarted for camera Camera1.
2024-12-12 11:27:11,731 [ERROR]: NVR: Recording for camera Camera2 has stopped unexpectedly!
2024-12-12 11:27:11,732 [ERROR]: NVR: No ffmpeg process found for Camera2.
2024-12-12 11:27:11,734 [INFO]: NVR: Recording restarted for camera Camera2.
2024-12-12 11:27:11,744 [ERROR]: NVR: No connection to Camera1 at 10_10_10_215
2024-12-12 11:27:11,748 [ERROR]: NVR: No connection to Camera2 at 10_10_10_77

**Comment by cyb3rdoc** [comment]
I updated the code in dev branch to address your issue and docker image uses dev tag. Check dev docker image and let me know if this fixes your issue.

**Comment by WiZaR5789** [comment]
On latest dev container there is bug where logs looks like this and I encounter the same problem as the guy who opened this issue.

<img width="769" alt="image" src="https://github.com/user-attachments/assets/caccefd7-3c67-40ff-b9fd-9fc07011371f" />

**Comment by cyb3rdoc** [comment]
The 'log_debug' related bug has been fixed and updated in 'dev' branch and 'dev' docker image.

**Comment by WiZaR5789** [comment]
> The 'log_debug' related bug has been fixed and updated in 'dev' branch and 'dev' docker image.

Yes, I noticed, thanks a lot for the quick response to the problem. I'm currently testing the functionality being discussed in this thread. I'll let you know if it has been resolved.

**Comment by WiZaR5789** [comment]
> I updated the code in dev branch to address your issue and docker image uses dev tag. Check dev docker image and let me know if this fixes your issue.

The problem seems to be resolved, no duplicate recordings for over 6 hours. Great, simple recorder. Thanks again for the quick fixes!
